### PR TITLE
Fixes #10867 - OpenStack boot_from_volume is true only when checked

### PR DESCRIPTION
**Pull request by dLobatog**

### CODE REVIEW

File: test/unit/compute_resources/openstack_test.rb [hunk 1]
```diff
@@ -1,10 +1,30 @@
...
+    assert_equal host, as_admin { @compute_resource.associated_host(iface) }
+  end
+
+  test "boot_from_volume does not get triggered when a string 'false' is passed as argument" do
+    Fog.mock!
+    @compute_resource.stubs(:key_pair).returns(mocked_key_pair)
+    @compute_resource.expects(:boot_from_volume).times(0)
+    @compute_resource.create_vm(:boot_from_volume => 'false', :nics => [""],
+                                :flavor_ref => 'foo_flavor', :image_ref => 'foo_image')
+    Fog.unmock!
```
Comment (domcleal) [inline test/unit/compute_resources/openstack_test.rb]: Put this into a teardown if possible, so it remains unmocked if an assertion is raised.

File: test/unit/compute_resources/openstack_test.rb [hunk 2]
```diff
@@ -1,10 +1,30 @@
...
-    cr = FactoryGirl.build(:openstack_cr)
     iface = mock('iface1', :floating_ip_address => '10.0.0.154', :private_ip_address => "10.1.1.1")
-    assert_equal host, as_admin { cr.associated_host(iface) }
+    assert_equal host, as_admin { @compute_resource.associated_host(iface) }
+  end
+
+  test "boot_from_volume does not get triggered when a string 'false' is passed as argument" do
+    Fog.mock!
+    @compute_resource.stubs(:key_pair).returns(mocked_key_pair)
+    @compute_resource.expects(:boot_from_volume).times(0)
```
Comment (domcleal) [inline test/unit/compute_resources/openstack_test.rb]: Nitpick, s/times(0)/never/

**Comment by domcleal** [comment]
If somebody can test that this works (only boots from volume if ticked), I'd appreciate it, I don't have a working CR.

Edit: @shlomizadok has offered

**Comment by stbenjam** [comment]
Not testable for me because of  http://projects.theforeman.org/issues/10969

**Comment by shlomizadok** [comment]
tested, works as expected

**Comment by domcleal** [comment]
Merged as 776cff9, thanks @eLobato.

**Comment by domcleal** [comment]
Thanks for the test too @shlomizadok!

**Comment by AtzeDeVries** [comment]
I'm not sure it is really a bug but i've replaced this openstack.rb with the original on our foreman 1.8.2 (on ubuntu 14.04). This results in the follow error

2015-07-28 10:56:33 [I] Successfully decrypted field for Foreman::Model::Openstack DevOps
2015-07-28 10:56:34 [W] Failed to create a compute DevOps (OpenStack) instance testur.openstacklocal: uninitialized constant Foreman::Cast
 /usr/share/foreman/app/models/compute_resources/foreman/model/openstack.rb:93:in `rescue in create_vm'
/usr/share/foreman/app/models/compute_resources/foreman/model/openstack.rb:77:in`create_vm'
/usr/share/foreman/app/models/concerns/orchestration/compute.rb:76:in `setCompute'
/usr/share/foreman/app/models/concerns/orchestration.rb:141:in`execute'
/usr/share/foreman/app/models/concerns/orchestration.rb:86:in `block in process'
/usr/share/foreman/app/models/concerns/orchestration.rb:78:in`each'
/usr/share/foreman/app/models/concerns/orchestration.rb:78:in `process'
/usr/share/foreman/app/models/concerns/orchestration.rb:19:in`on_save'
/usr/share/foreman/vendor/ruby/1.9.1/gems/activesupport-3.2.21/lib/active_support/callbacks.rb:649:in `_run__1750397551040917370__save__2678967843180454136__callbacks'
/usr/share/foreman/vendor/ruby/1.9.1/gems/activesupport-3.2.21/lib/active_support/callbacks.rb:405:in`__run_callback'
/usr/share/foreman/vendor/ruby/1.9.1/gems/activesupport-3.2.21/lib/active_support/callbacks.rb:385:in `_run_save_callbacks'
/usr/share/foreman/vendor/ruby/1.9.1/gems/activesupport-3.2.21/lib/active_support/callbacks.rb:81:in`run_callbacks'
/usr/share/foreman/vendor/ruby/1.9.1/gems/activerecord-3.2.21/lib/active_record/callbacks.rb:264:in `create_or_update'
/usr/share/foreman/vendor/ruby/1.9.1/gems/activerecord-3.2.21/lib/active_record/persistence.rb:84:in`save'
/usr/share/foreman/vendor/ruby/1.9.1/gems/activerecord-3.2.21/lib/active_record/validations.rb:50:in `save'
/usr/share/foreman/vendor/ruby/1.9.1/gems/activerecord-3.2.21/lib/active_record/attribute_methods/dirty.rb:22:in`save'
/usr/share/foreman/vendor/ruby/1.9.1/gems/activerecord-3.2.21/lib/active_record/transactions.rb:259:in `block (2 levels) in save'
/usr/share/foreman/vendor/ruby/1.9.1/gems/activerecord-3.2.21/lib/active_record/transactions.rb:313:in`block in with_transaction_returning_status'
/usr/share/foreman/vendor/ruby/1.9.1/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract/database_statements.rb:192:in `transaction'
/usr/share/foreman/vendor/ruby/1.9.1/gems/activerecord-3.2.21/lib/active_record/transactions.rb:208:in`transaction'
/usr/share/foreman/vendor/ruby/1.9.1/gems/activerecord-3.2.21/lib/active_record/transactions.rb:311:in `with_transaction_returning_status'
/usr/share/foreman/vendor/ruby/1.9.1/gems/activerecord-3.2.21/lib/active_record/transactions.rb:259:in`block in save'
/usr/share/foreman/vendor/ruby/1.9.1/gems/activerecord-3.2.21/lib/active_record/transactions.rb:270:in `rollback_active_record_state!'
/usr/share/foreman/vendor/ruby/1.9.1/gems/activerecord-3.2.21/lib/active_record/transactions.rb:258:in`save'

**Comment by domcleal** [comment]
It uses another feature that's only in 1.9 or higher, you could probably add this file to your installation: https://github.com/theforeman/foreman/blob/develop/lib/foreman/cast.rb (or wait for 1.8.3)

**Comment by AtzeDeVries** [comment]
Ok thnx, i just commented line 77 (in https://github.com/eLobato/foreman/blob/d7ad9f58fdc60de98381c274b1c75ac794155302/app/models/compute_resources/foreman/model/openstack.rb). We won't be using boot from volume.

**Pull request closed without merging**